src/CBLDatabase.cc

```diff
@@ -258,60 +198,31 @@ bool CBLDatabase::deleteCollection(slice collectionName, slice scopeName) {
     
     auto spec = C4Database::CollectionSpec(collectionName, scopeName);
     c4db->deleteCollection(spec);
-    removeCBLCollection(spec);
     return true;
 }
 
 
 Retained<CBLScope> CBLDatabase::getDefaultScope() {
-    _c4db->useLocked();
     return getScope(kC4DefaultScopeID);
 }
 
 
-Retained<CBLCollection> CBLDatabase::getDefaultCollection(bool mustExist) {
-    auto db = _c4db->useLocked();
-    
-    if (_defaultCollection && !_defaultCollection->isValid()) {
-        _defaultCollection = nullptr;
-    }
-    
-    if (!_defaultCollection && mustExist) {
-        C4Error::raise(LiteCoreDomain, kC4ErrorNotOpen,
-                       "Invalid collection: either deleted, or db closed");
-    }
-    
-    return _defaultCollection;
+Retained<CBLCollection> CBLDatabase::getDefaultCollection() {
+    return getCollection(kC4DefaultCollectionName, kC4DefaultScopeID);
 }
 
 
-Retained<CBLCollection> CBLDatabase::createCBLCollection(C4Collection* c4col, CBLScope* scope, bool cache) {
-    auto retainedCollection = make_retained<CBLCollection>(c4col, scope, const_cast<CBLDatabase*>(this));
-    auto collection = retainedCollection.get();
-    if (cache) {
-        _collections.insert({C4Database::CollectionSpec(c4col->getSpec()), std::move(retainedCollection)});
-    }
-    return collection;
+Retained<CBLCollection> CBLDatabase::createCBLCollection(C4Collection* c4col, CBLScope* scope) {
+    return new CBLCollection(c4col, scope, const_cast<CBLDatabase*>(this));
```

Comment:
Would it be possible to have this instead of the every call of `createCBLCollection` for both `getCollection` and `createCollection`, similar to Scope implementation, having returning `new`. Would that be a bit more readable?

It won't make much of a difference... I might just be picky